### PR TITLE
build: make stacktrace configureable

### PR DESCRIPTION
**Pull request by travankor**
This should help users that want coredump generation to work correctly.

Systemd-coredump should work when this option is disabled, but I haven't
tested systemd-coredump specifically.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ static void wlr_log_handler(wlr_log_importance level,
     wf::log::log_plain(wlevel, buffer);
 }
 
-#ifndef ASAN_ENABLED
+#ifndef ASAN_ENABLED || COREDUMP_ENABLED
```
Comment (ammen99) [inline src/main.cpp:119]: is ifndef the appropriate thing to use here?

**Review by ammen99 — CHANGES_REQUESTED** [review]
I'd rather have a single option which controls stracktrace, not one for asan and one for coredump, it seems confusing to me, especially because both are used for the same thing.

File: src/meson.build [hunk 2]
```diff
@@ -61,9 +61,12 @@ endif
 
 
 cxx_flags_asan = run_command('/bin/sh', '-c', 'echo $CXXFLAGS $CPPFLAGS | grep fsanitize')
-if get_option('b_sanitize').contains('address') or cxx_flags_asan.returncode() == 0
-  message('Address sanitizer enabled, disabling internal backtrace')
-  debug_arguments += ['-DASAN_ENABLED']
+if (not (get_option('b_sanitize').contains('address') or cxx_flags_asan.returncode() == 0)
```
Comment (ammen99) [inline src/meson.build:64]: I find this a bit confusing, took me awhile to understand what the ifs do exactly. Can't we just set `print_trace` to false if asan is enabled, and then add the debug arguments in a second if?

**Review by ammen99 — CHANGES_REQUESTED** [review]

File: src/meson.build (hunk 2 above)
Comment (travankor) [reply src/meson.build]: Sorry for the delay

**Review by ammen99 — APPROVED** [review]
LGTM now, thanks!

**Pull request merged**